### PR TITLE
Standardize logging - many places

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
As part of the series of PRs to standardize logging (#4631, #4576), here is the next batch

Here we update:
 - `bin/hwinj`
 - `bin/live`
 - `bin/population`
 - `bin/minifollowup`
 - `pycbc/catalog`
 - `pycbc/coordinates`
 - `pycbc/distributions`
 - `pycbc/frame`
 - `pycbc/workflow`

### CODE REVIEW

**Comment by GarethCabournDavies** [comment]
Note the remaining codeclimate issue is not one introduced here, it simply that has a logging statement nearby

File: bin/population/pycbc_population_plots [hunk 1]
```diff
@@ -27,19 +27,23 @@ __date__ = "31.10.2017"
...
+import pycbc
+import pycbc.version
 from pycbc.population import rates_functions as rf
-from numpy import logaddexp, log, newaxis, expm1
 
 # Parse command line
-parser = argparse.ArgumentParser(description=
-                                      "Plot rate prior and posteriors.")
+parser = argparse.ArgumentParser(
+    description="Plot rate prior and posteriors."
```
Comment (titodalcanton) [inline bin/population/pycbc_population_plots:44]: Use `__doc__` here and put the description in the docstring.

File: pycbc/workflow/core.py [hunk 2]
```diff
@@ -486,7 +500,7 @@ def update_current_retention_level(self, value):
             msg="Cannot find file-retention-level in [workflow] section "
             msg+="of the configuration file. Setting a default value of "
             msg+="retain all files."
-            logging.warn(msg)
+            logger.warn(msg)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:503]: `warn()` is deprecated, use `warning()`.

File: pycbc/workflow/core.py [hunk 3]
```diff
@@ -517,7 +531,7 @@ def update_current_retention_level(self, value):
                     warn_msg += "been set in class {0}. ".format(type(self))
                     warn_msg += "This value should be set explicitly. "
                     warn_msg += "All output from this class will be stored."
-                    logging.warn(warn_msg)
+                    logger.warn(warn_msg)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:534]: Use `warning()`.

File: pycbc/workflow/core.py [hunk 4]
```diff
@@ -540,7 +554,7 @@ def update_current_tags(self, tags):
         if tags is None:
             tags = []
         if '' in tags:
-            logging.warn('DO NOT GIVE ME EMPTY TAGS (in %s)', self.name)
+            logger.warn('DO NOT GIVE ME EMPTY TAGS (in %s)', self.name)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:557]: Use `warning()`.

File: pycbc/workflow/core.py [hunk 5]
```diff
@@ -549,7 +563,7 @@ def update_current_tags(self, tags):
             warn_msg = "This job has way too many tags. "
             warn_msg += "Current tags are {}. ".format(' '.join(tags))
             warn_msg += "Current executable {}.".format(self.name)
-            logging.info(warn_msg)
+            logger.info(warn_msg)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:566]: Maybe this should be `warning()` instead of `info()`?

File: pycbc/workflow/core.py [hunk 6]
```diff
@@ -595,7 +609,7 @@ def update_current_tags(self, tags):
             else:
                 warn_string = "warning: config file is missing section "
                 warn_string += "[{0}]".format(sec)
-                logging.warn(warn_string)
+                logger.warn(warn_string)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:612]: `warning()`.

File: pycbc/workflow/core.py [hunk 7]
```diff
@@ -1152,7 +1166,7 @@ def __init__(self, ifos, exe_name, segs, file_url=None,
         if tags is None:
             tags = []
         if '' in tags:
-            logging.warn('DO NOT GIVE EMPTY TAGS (from %s)', exe_name)
+            logger.warn('DO NOT GIVE EMPTY TAGS (from %s)', exe_name)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:1169]: `warning()`.

File: pycbc/workflow/core.py [hunk 8]
```diff
@@ -1832,7 +1846,7 @@ def from_segment_list_dict(cls, description, segmentlistdict,
                         # Setting valid segment now is hard!
                         warn_msg = "No information with which to set valid "
                         warn_msg += "segment."
-                        logging.warn(warn_msg)
+                        logger.warn(warn_msg)
```
Comment (titodalcanton) [inline pycbc/workflow/core.py:1849]: `warning()`

File: pycbc/workflow/datafind.py [hunk 9]
```diff
@@ -830,7 +832,7 @@ def get_missing_segs_from_frame_file_cache(datafindcaches):
             if not cache[0].scheme == 'file':
                 warn_msg = "We have %s entries in the " %(cache[0].scheme,)
                 warn_msg += "cache file. I do not check if these exist."
-                logging.info(warn_msg)
+                logger.info(warn_msg)
```
Comment (titodalcanton) [inline pycbc/workflow/datafind.py:835]: This should probably use `warning()`.

File: pycbc/workflow/datafind.py [hunk 10]
```diff
@@ -1040,7 +1042,7 @@ def log_datafind_command(observatory, frameType, startTime, endTime,
         else:
             errMsg = "Unknown datafind kwarg given: %s. " %(name)
             errMsg+= "This argument is stripped in the logged .sh command."
-            logging.warn(errMsg)
+            logger.warn(errMsg)
```
Comment (titodalcanton) [inline pycbc/workflow/datafind.py:1045]: `warning()`

File: pycbc/workflow/psdfiles.py [hunk 11]
```diff
@@ -136,7 +139,7 @@ def setup_psd_pregenerated(workflow, tags=None):
             except ConfigParser.Error:
                 # It's unlikely, but not impossible, that only some ifos
                 # will have pregenerated PSDs
-                logging.warn("No psd file specified for IFO %s." % (ifo,))
+                logger.warn("No psd file specified for IFO %s.", ifo)
```
Comment (titodalcanton) [inline pycbc/workflow/psdfiles.py:142]: `warning()`

**Comment by titodalcanton** [comment]
Some comments (which maybe you want to keep to a separate PR) but otherwise this looks good.

**Comment by GarethCabournDavies** [comment]
I've updated the highlighted places to use warning instead of warn.

I thought I'd sort all places to do this as well while thinking of it, so `git grep "warn("` now only shows things from the warnings module

File: bin/population/pycbc_population_plots (hunk 1 above)
Comment (GarethCabournDavies) [reply bin/population/pycbc_population_plots]: used __doc__, the info was already in the docstring though

File: bin/bank/pycbc_geom_aligned_2dstack [hunk 12]
```diff
@@ -269,7 +271,9 @@ for entry in temp_bank:
                     vec3_max = temmpvec3_max
                     vec3_depth = vec3_max - vec3_min
             if vec3_max < xis_close[:,2].max():
-                logging.warn("WARNING: Numerical placement fails, trying again")
+                logging.warniing(
```
Comment (titodalcanton) [inline bin/bank/pycbc_geom_aligned_2dstack]: Tyypo!

Comment (titodalcanton) [reply bin/bank/pycbc_geom_aligned_2dstack]: Ah, already fixed.

**Review by titodalcanton — APPROVED** [review]
Looks good.

**Pull request merged**